modules/swiss/src/main/SwissApi.scala

```diff
@@ -315,11 +315,10 @@ final class SwissApi(
     }
 
   private def getGameRanks(swiss: Swiss, game: Game): Fu[Option[GameRanks]] =
-    game.whitePlayer.userId.ifTrue(swiss.isStarted) so { whiteId =>
-      game.blackPlayer.userId.so: blackId =>
+    swiss.isStarted.so:
+      game.players.traverse(_.userId).so: ids =>
         rankingApi(swiss).map: ranking =>
-          (ranking.get(whiteId), ranking.get(blackId)) mapN GameRanks.apply
-    }
+          ids.traverse(ranking.get).map(_.reduce(GameRanks.apply))
```

Comment:
Another way to write this is: `ids.map(ranking.get).mapN(GameRanks.apply)`

Or maybe we should introduce:

```scala
  def traverseReduce[F[_], B, C](f: A => F[B])(r: (B, B) => C): Applicative[F] ?=> F[C] =
    (f(white), f(black)).mapN(r)
```
and rewrite the line above as:
`ids.traverseReduce(ranking.get)(GameRanks.apply)`